drools-wb-webapp/pom.xml

```diff
@@ -552,46 +444,93 @@
     </dependency>
 
     <dependency>
-      <groupId>org.uberfire</groupId>
-      <artifactId>uberfire-social-activities-client</artifactId>
+      <groupId>org.kie.workbench.screens</groupId>
+      <artifactId>kie-wb-common-social-home-page-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.kie.workbench.screens</groupId>
+      <artifactId>kie-wb-common-social-home-page-backend</artifactId>
     </dependency>
 
     <dependency>
       <groupId>org.kie.workbench.screens</groupId>
-      <artifactId>kie-wb-common-social-home-page-api</artifactId>
+      <artifactId>kie-wb-common-examples-screen-api</artifactId>
     </dependency>
     <dependency>
       <groupId>org.kie.workbench.screens</groupId>
-      <artifactId>kie-wb-common-social-home-page-client</artifactId>
-      <scope>provided</scope>
+      <artifactId>kie-wb-common-examples-screen-backend</artifactId>
     </dependency>
+
     <dependency>
       <groupId>org.kie.workbench.screens</groupId>
-      <artifactId>kie-wb-common-social-home-page-backend</artifactId>
+      <artifactId>kie-wb-common-workbench-backend</artifactId>
     </dependency>
 
     <dependency>
       <groupId>org.kie.workbench.screens</groupId>
-      <artifactId>kie-wb-common-examples-screen-api</artifactId>
+      <artifactId>kie-wb-common-library-backend</artifactId>
     </dependency>
+
     <dependency>
       <groupId>org.kie.workbench.screens</groupId>
-      <artifactId>kie-wb-common-examples-screen-client</artifactId>
-      <scope>provided</scope>
+      <artifactId>kie-wb-common-library-api</artifactId>
     </dependency>
+    <!-- Ugly, but some generated code depends on classes from client modules...
+     - is this really needed on server? i.e. is the generated class (org/jboss/errai/cdi/EventQualifierSerializerImpl.class) ever loaded?
+       - why I have never seen class loading issues even though the client artifacts weren't in WEB-INF/lib?
+     - could/should these dependencies (annotations, interfaces) be moved from -client modules, to -api modules?
+     - or should the generated classes be excluded from WEB-INF/classes? I can imagine they're generated
+       during GWT compilation, then compiled to JavaScript and never used on server
+    -->
```

Comment:
Some of the following generated classes depend on client artifacts. The classes are generated by Errai during GWT compilation. My question is: are these classes just a by-product in the process of generating a Java source, compiling it into bytecode and finally transpiling into JS (although I don't know if transpilation into JS requires classes or Java sources), and thus should not be packaged inside the web app WAR file? Or, since they concern serialization and marshalling, are they used by backend too and should be there? I think second option makes sense but what I'm confused about is that everything seems to be working even if the client artifacts are not present in `WEB-INF/lib`. I would expect a `NoClassDefFoundError` sooner or later.

Can someone help me understand this? Maybe @ederign?

```
target/drools-wb-webapp/WEB-INF/classes/org/jboss/errai/
├── cdi
│   ├── EventQualifierSerializerImpl$10.class
│   ├── EventQualifierSerializerImpl$11.class
│   ├── EventQualifierSerializerImpl$12.class
│   ├── EventQualifierSerializerImpl$1.class
│   ├── EventQualifierSerializerImpl$2.class
│   ├── EventQualifierSerializerImpl$3.class
│   ├── EventQualifierSerializerImpl$4.class
│   ├── EventQualifierSerializerImpl$5.class
│   ├── EventQualifierSerializerImpl$6.class
│   ├── EventQualifierSerializerImpl$7.class
│   ├── EventQualifierSerializerImpl$8.class
│   ├── EventQualifierSerializerImpl$9.class
│   └── EventQualifierSerializerImpl.class
├── ServerMarshallingFactoryImpl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_j_l_StackTraceElement_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_j_l_String_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_o_d_w_m_d_r_ActionFieldValue_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_o_d_w_m_d_r_ConnectiveConstraint_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_o_d_w_m_d_r_FieldConstraint_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_o_d_w_m_d_r_IAction_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_o_d_w_m_d_r_IFactPattern_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_o_d_w_m_d_r_IPattern_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_o_d_w_m_d_r_RuleAttribute_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_o_d_w_m_d_r_RuleMetadata_D1_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_a_o_d_w_m_t_s_CallFieldValue_D1_Impl.class
... (lines skipped) ...
├── ServerMarshallingFactoryImpl$Marshaller_o_u_w_m_i_ContextDefinitionImpl_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_o_u_w_m_i_PanelDefinitionImpl_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_o_u_w_m_i_PartDefinitionImpl_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_o_u_w_m_i_PerspectiveDefinitionImpl_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_o_u_w_m_i_SplashScreenFilterImpl_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_o_u_w_m_NamedPosition_Impl.class
├── ServerMarshallingFactoryImpl$Marshaller_o_u_w_m_Notification_erraiD_State_Impl.class
└── ServerMarshallingFactoryImpl$Marshaller_o_u_w_m_Notification_Impl.class

1 directory, 791 files
```

Reply:
@yurloc I think they should be in WEB-INF/classes (for example; the marshalling class for ```FieldConstraint``` would be used when opening a Guided Rule on the server and marshalling it to the browser - along with a lot of the other classes listed above). What I do not know however is whether Errai falls back to dynamic marshalling" as opposed to using "static marshalling" IF these classes are missing.